src/nabit/bin/cli.py

```diff
@@ -152,9 +152,76 @@ def archive(
             if isinstance(task, PathCollectionTask):
                 task.hard_links = True
 
+    signatures = _get_signatures(ctx.raw_args, signature_args)
```

Comment:
I pulled the logic for massaging the command line arguments into signatures into a function _get_signatures() since both `archive` and `sign` needed to do it.